go/dpagg/count.go

```diff
@@ -194,6 +195,9 @@ func (c *Count) ThresholdedResult(thresholdDelta float64) *int64 {
 //
 // See https://github.com/google/differential-privacy/tree/main/common_docs/confidence_intervals.md.
 func (c *Count) ComputeConfidenceInterval(alpha float64) (noise.ConfidenceInterval, error) {
+	if c.isSerialized {
```

Comment:
I think here we should only check resultReturned (if it was serialized then it automatically means that there was no result returned).

Also I think we need more general flag: isUsable, isUsed, smth like that (I don't have a strong opinion about the name).

Why:

we also have merge method. It sets resultReturned to true as well, it will break confidence intervals.
We could either add third flag (isMerge), or make one general flag which signals whether this instance can be used in computations or not.

And we need to check this flag (along with resultReturned) in Result(), Merge() [e,g see checkMergeCount] and Serialize/encode.

The same for Java.
And we need to consider all possible combinations of calls of methods in tests, to cover as much as possible.